src/plugins/emojis.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * SCEditor Paragraph Formatting Plugin
+ * http://www.sceditor.com/
+ *
+ * Copyright (C) 2011-2024, Sam Clarke (samclarke.com)
+ *
+ * SCEditor is licensed under the MIT license:
+ *	http://www.opensource.org/licenses/mit-license.php
+ *
+ * @fileoverview SCEditor Paragraph Formatting Plugin
+ * @author Sam Clarke
+ */
+
+(function(sceditor) {
+    'use strict';
+
+    sceditor.plugins.emojis = function() {
+        const base = this;
+
+        /**
+         * Function for the exec and txtExec properties
+         *
+         * @param  {node} caller
+         * @private
+         */
+        var emojisCmd = function(caller) {
+            const editor = this,
+                content = document.createElement('div'),
+                emojis = editor.opts.emojis || [],
+                perLine = Math.sqrt(Object.keys(emojis).length);
+
+            if (!emojis.length) {
+                const pickerOptions = { onEmojiSelect: handleSelect };
+                const picker = new EmojiMart.Picker(pickerOptions);
```

Comment:
What is EmojiMart? This will cause an error if it isn't defined. If it's falling back to an external library it might be worth adding a check to make sure it's included.